src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -218,42 +215,39 @@ public synchronized RcaResponse readRca(String rca) {
-        .buildSummaryQuery(create, currLevelTable, upperLevelPrimaryKey, foreignKeyField);
-    try {
-      Result<Record> recordList = rcaQuery.fetch();
-      for (Record record : recordList) {
-        GenericSummary summary = null;
-        if (upperLevelSummary instanceof RcaResponse) {
-          summary = HotClusterSummary.buildSummary(record);
-        }
-        else if (upperLevelSummary instanceof HotClusterSummary) {
-          summary = HotNodeSummary.buildSummary(record);
-        }
-        else if (upperLevelSummary instanceof HotNodeSummary) {
-          summary = HotResourceSummary.buildSummary(record);
-        }
-        else if (upperLevelSummary instanceof HotResourceSummary) {
-          summary = TopConsumerSummary.buildSummary(record);
-        }
-        if (summary != null) {
-          Field<Integer> primaryKeyField = DSL.field(
-              SQLiteQueryUtils.getPrimaryKeyColumnName(summary.getTableName()), Integer.class);
-          readSummary(summary, record.get(primaryKeyField));
-          upperLevelSummary.addNestedSummaryList(summary);
+    for (Class<? extends GenericSummary> clazz : clazzList) {
+      Field<Integer> foreignKeyField = DSL.field(
+          SQLiteQueryUtils.getPrimaryKeyColumnName(upperLevelTable), Integer.class);
+      SelectJoinStep<Record> rcaQuery = SQLiteQueryUtils
+          .buildSummaryQuery(create, clazz.getSimpleName(), upperLevelPrimaryKey, foreignKeyField);
+      try {
+        Result<Record> recordList = rcaQuery.fetch();
+        for (Record record : recordList) {
+          Method method = clazz.getMethod("buildSummary", Record.class);
+          GenericSummary summary = (GenericSummary) method.invoke(null, record);
+          if (summary != null) {
+            Field<Integer> primaryKeyField = DSL.field(
+                SQLiteQueryUtils.getPrimaryKeyColumnName(summary.getTableName()), Integer.class);
+            readSummary(summary, record.get(primaryKeyField));
+            upperLevelSummary.addNestedSummaryList(summary);
+          }
         }
       }
-    }
-    catch (DataAccessException de) {
-      // it is totally fine if we fail to read some certain tables as some types of summaries might be missing
-      LOG.warn("Fail to read Summary table : {}, query = {},  exceptions : {}", currLevelTable, rcaQuery.toString(), de.getStackTrace());
+      catch (DataAccessException de) {
+        // it is totally fine if we fail to read some certain tables as some types of summaries might be missing
+        LOG.warn("Fail to read Summary table : {}, query = {},  exceptions : {}",
+            clazz.getSimpleName(), rcaQuery.toString(), de.getStackTrace());
+      } catch (Exception e) {
+        // we might got a reflection issue if running into this. Check the NestedTableMap and make sure
+        // the summary class has been added.
+        LOG.error("Fail to use reflection to build GenericSummary, trace = {}", e.getStackTrace());
```

Comment:
Consider adding a metric here.

Reply:
I don't think we need to add a metric here. we will only run into this if there is a bug in this code which doesn't needs to be shown up in the metric.